quinn-proto/src/endpoint.rs

```diff
@@ -109,6 +109,11 @@ where
         self.transmits.pop_front()
     }
 
+    /// Replace the server configuration, affecting new incoming connections only
+    pub fn set_server_config(&mut self, server_config: Option<Arc<ServerConfig<S>>>) {
```

Comment:
Why allow `Option`? For servers, this must always be `Some`.

Reply:
Reviewing the code, it looks to me like this will allow enabling/disabling server operation at will, which seems somewhere between harmless and useful, and less surprising than silently doing nothing. I guess we could omit the `Option` and let it be possible to enable server mode but not disable it, but why not be symmetrical?